src/extensions/savetoLaserfiche/SaveToLaserficheDialog.tsx

```
@@ -81,23 +91,71 @@ function SaveToLaserficheDialog(props: {
-          const saveToLF = new SaveDocumentToLaserfiche(props.spFileMetadata);
+          const validRepoClient = await tryGetValidRepositoryClientAsync();
+          const saveToLF = new SaveDocumentToLaserfiche(
+            props.spFileMetadata,
+            validRepoClient
+          );
           try {
-            const successSaveToLF =
-              await saveToLF.trySaveDocumentToLaserficheAsync();
-            props.isSuccessfulLoggedIn(true);
-            setSuccess(successSaveToLF);
+            try {
+              const repoId = await validRepoClient.getCurrentRepoId();
+              const entryInfo: Entry =
+                await validRepoClient.entriesClient.getEntry({
+                  repoId,
+                  entryId: Number.parseInt(props.spFileMetadata.entryId, 10),
+                });
+              const name = validRepoClient.entriesClient.getEntryByPath({
+                repoId,
+                fullPath: PathUtils.combinePaths(
+                  entryInfo.fullPath,
+                  props.spFileMetadata.fileName
+                ),
+              });
+              if (name) {
+                setShowSaveTo(false);
+                const status = await getConfirmation(
+                  ENTRY_WITH_SAME_NAME_EXISTS_IN_FOLDER_IF_CONTINUE_LF_WILL_RENAME
+                );
+                if (status) {
+                  setShowSaveTo(true);
+                  await continueSavingDocumentAsync(saveToLF);
+                } else {
+                  window.localStorage.removeItem(SP_LOCAL_STORAGE_KEY);
+                  props.isSuccessfulLoggedIn(true);
+                  await props.closeClick();
+                }
+              }
+            } catch (err) {
+              const docDoesNotAlreadyExists = (err as APIServerException).statusCode === 404;
+              if (docDoesNotAlreadyExists) {
+                await continueSavingDocumentAsync(saveToLF);
+              }
+            }

```

afterwards we should probably throw an error just in case it's not a 404